cosign/README.md

```diff
@@ -0,0 +1,247 @@
+
+1. Introduce an OpenID Connect (OIDC) Provider, assure that a client is in control of an identity
+2. Issue short-lived public-key certificates with these identities (from a Certificate Authority), and publish these certificates to an Identity Log.
+3. Publish a signature over an artifact (or artifact meta-data) to an Artifact Log, allowing verifiers to check its validity.
+
+The first and second operations are performed by a system called **Fulcio**, which functions as a Certificate Authority and transparency log for a namespace of OIDC identities (Identity Log). The third operation is performed by a system called **Rekor**, a transparency log for artifact signatures (Artifact Log). And **Cosign** is the reference Sigstore client implementation to sign arbitrary artifacts. With these components, Sigstore completed trust setup, signing and verification.
+
+### Sign Workflow
+
+Following graph explains how general signing works in Sigstore. Fulcio issues short-lived certificates binding an ephemeral key to an OpenID Connect identity. Signing events are logged in Rekor, a signature transparency log, providing an auditable record of when a signature was created.
+Sigstore’s root of trust, which includes Fulcio’s root CA certificate and Rekor’s public key, are distributed by The Update Framework (TUF).
+
+```mermaid
+flowchart TD
+    subgraph Sigstore Trust Infrastructure
+        TUF[Root Key Ceremony] -->|Establishes| TR[Sigstore Trust Root]
+        TR -->|Manages Entry| CA[Fulcio]
+        TR -->|Manages Entry| REKOR[Rekor]
+    end
+
+    subgraph Signing Workflow
+        SU(EndUser) -->|1.Sign Input| SR[Cosign Signer]
+        SR -->|2.Authenticates with| OIDC[OIDC Provider]
+        OIDC -->|3.Issues Token| SR
+        SR --->|4.Requests Certificate| CA
+        CA -->|5.Records Certificate| REKOR
+        CA -->|6.Issues Certificate| SR
+        SR -->|8.Provides Signing Result| SU
+        
+    end
+
+    subgraph OCI Distribution
+        SR --->|7.Signs & Publishes| REG[OCI Registry]
+    end
+
+    style SU fill:#A7C6DA, color: #000000
+```
+
+### Verify Workflow
+
+Sigstore compares a tuple of signature, key/certificate, and artifact from the timestamped object against the timestamped Rekor entry. If they match, it confirms that the signature is valid because the user knows that the expected software creator, whose identity was certified at the moment of signing, published the software artifact in their possession.
+
+```mermaid
+flowchart TD
+    subgraph Sigstore Trust Infrastructure
+        TUF[Root Key Ceremony] -->|Establishes| TR[Sigstore Trust Root]
+        TR -->|Manages Entry| CA[Fulcio]
+        TR -->|Manages Entry| REKOR[Rekor]
+    end
+    subgraph Verification Workflow
+        VU(EndUser) -->|1.Verify Input|VR[Cosign Verifier]
+        VR -->|7.Requests Log Proof| REKOR
+        VR ---->|4.Requests Root Certificates| TR
+        TR ---->|5.Provides Root Certificates| VR
+        REKOR -->|6.Provides Proof| VR 
```

Comment:
Can you provide evidence on the steps 6 and 7?

Reply:
https://github.com/sigstore/cosign/blob/main/cmd/cosign/cli/verify/verify.go#L174

Reply:
It just show that it gets the public keys of the Rekor (i.e. `rekor.pub`).

Reply:
https://github.com/sigstore/cosign/blob/main/pkg/cosign/verify.go#L778

Reply:
Here is how cosign perform the verification against the cosign transparency log: https://github.com/sigstore/cosign/blob/main/pkg/cosign/verify.go#L530

Reply:
In the latest implementation from `sigstore-go`, check transparency log is a default behavior.
Ref: https://github.com/sigstore/sigstore-go/blob/main/pkg/verify/signed_entity.go#L578